sklearn/preprocessing/_discretization.py

```diff
@@ -184,6 +189,7 @@ def fit(self, X, y=None):
         n_bins = self._validate_n_bins(n_features)
 
         bin_edges = np.zeros(n_features, dtype=object)
+        sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
```

Comment:
Please move this line directly after `X = self._validate_data(X, dtype="numeric")` and put it in an if condition: `if sample_weight is not None:`.

Reply:
Code refactored.